dnshostprovider.go

```diff
@@ -2,10 +2,24 @@ package zk
 
 import (
 	"fmt"
+	"context"
 	"net"
 	"sync"
+	"time"
 )
 
+const _defaultLookupTimeout = 3 * time.Second
+
+// DNSHostProviderOption is an option for the DNSHostProvider.
+type DNSHostProviderOption func (*DNSHostProvider)
```

Comment:
nit: i would like to start by using an options struct https://github.com/uber-go/guide/blob/master/style.md#functional-options

this is just one option so either way works

Reply:
conn.go builds options with function types but this allows outside packages to create their own options.

Changed it to the struct + interface options as per styleguide.